x-pack/platform/packages/shared/onechat/onechat-common/tools/tool_ids.ts

```diff
@@ -5,21 +5,15 @@
  * 2.0.
  */
 
-import { builtInToolIdPrefix, reservedKeywords } from './constants';
+export const toolIdRegexp =
+  /^(?:[a-z0-9](?:[a-z0-9_-]*[a-z0-9])?)(?:\.(?:[a-z0-9](?:[a-z0-9_-]*[a-z0-9])?))*$/;
 
-export const idRegexp = /^[a-z0-9](?:[a-z0-9_-]*[a-z0-9])?$/;
-
-/**
- * Check if the given ID is a built-in ID (starting with `.`)
- */
-export const isBuiltInToolId = (id: string) => {
-  return id.startsWith(builtInToolIdPrefix);
-};
+const reservedKeywords = ['new'];
 
 /**
  * Check if the given ID is a reserved ID
- * Atm this only checks for built-in IDs, but it will check for MCP and such later.
+ * Atm this only checks for `new` because that's a value we're using for url paths on the UI.
  */
 export const isReservedToolId = (id: string) => {
-  return isBuiltInToolId(id) || reservedKeywords.includes(id);
+  return reservedKeywords.includes(id);
```

Comment:
I discovered that - we are blocking tools named `new` because we use that in the url path for tool creation.

I don't really like that and we should address it ihmo - but that's out of the scope of this PR...

Reply:
Yeh something to keep in mind for documentation. Smells but ok for now.